### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by jacobweinstock**
## Description

This version provides improved logging and simplified server functions and methods. Also update all direct and transitive deps and add a dependabot config.

## Why is this needed

Fixes: #

## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade

### CODE REVIEW

**Comment by jacobweinstock** [comment]
Hey @chrisdoherty4, mind checking this one out when you have some cycles?

File: cmd/smee/main.go [hunk 1]
```diff
@@ -206,14 +208,27 @@ func main() {
...
+			if err != nil {
+				panic(fmt.Errorf("invalid tftp address for DHCP server: %w", err))
+			}
+			conn, err := server4.NewIPv4UDPConn("", net.UDPAddrFromAddrPort(bindAddr))
+			if err != nil {
+				panic(err)
+			}
+			defer func() {
+				_ = conn.Close()
+			}()
```
Comment (chrisdoherty4) [inline cmd/smee/main.go:228]: Does this throw a linter error when defering `conn.Close()` directly? If so, can we just use `//nolint: ...`?

**Review by chrisdoherty4 — DISMISSED** [review]
1 minor comment.

File: cmd/smee/main.go (hunk 1 above)
Comment (jacobweinstock) [reply cmd/smee/main.go:228]: its does. sure thing, will update.

Comment (jacobweinstock) [reply cmd/smee/main.go]: hmmm....actually i dont think the linting will complain. i must have just done it assuming the linter would complain. regardless, i've updated it.

**Comment by jacobweinstock** [comment]
Hey @chrisdoherty4. Sorry, couple little changes to this one since you last reviewed. Thanks!

**Review by chrisdoherty4 — APPROVED** [review]

**Pull request merged**